### PR TITLE
Update modules

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
Close https://github.com/nf-core/taxprofiler/issues/403

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/profiling.nf [hunk 1]
```diff
@@ -346,9 +346,8 @@ workflow PROFILING {
                                 .multiMap {
                                     it ->
                                         reads: [it[0] + it[2], it[1]]
-                                        db: it[3]
+                                        db: [ it[2], it[3] ]
                                 }
-
         // diamond only accepts single output file specification, therefore
         // this will replace output file!
         ch_diamond_reads_format = params.diamond_save_reads ? 'sam' : params.diamond_output_format
```
Comment (LilyAnderssonLee) [inline subworkflows/local/profiling.nf:353]: @jfy133  Do you know why we got this error when running `test_nonprocessing.config`? I have checked the `ch_input_for_diamond.reads` for `test.config` and `test_nonprocessing.config` and I can't see any difference between them except the reads of multiple runs were merged in the `test.config`.
```
[DUMP: diamond_test_reads] [['id':'2611', 'run_accession':'ERR5766174', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':true, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fasta/ERX5474930_ERR5766174_1.fa.gz]]
[DUMP: diamond_test_reads] [['id':'ERR3201952', 'instrument_platform':'OXFORD_NANOPORE', 'single_end':true, 'is_fasta':false, 'type':'long', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/home/lili.li/taxprofiler_update_tools_new/work/9f/ca4c96cee207ebba5e6d645ba7de20/ERR3201952_ERR3201952.unmapped_other.fastq.gz]]
[DUMP: diamond_test_reads] [['id':'2613', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/home/lili.li/taxprofiler_update_tools_new/work/f9/0c53ee2b2bccb1d4e1866a755962c8/2613_ERR5766181.unmapped.fastq.gz]]
[DUMP: diamond_test_reads] [['id':'2612', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/home/lili.li/taxprofiler_update_tools_new/work/b2/ed806ae5d462922f2c183229e9fce0/2612.merged.fastq.gz]]


[DUMP: diamond_test-noprocessing_reads] [['id':'2611', 'run_accession':'ERR5766174', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':true, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fasta/ERX5474930_ERR5766174_1.fa.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'2612', 'run_accession':'ERR5766176', 'instrument_platform':'ILLUMINA', 'single_end':false, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERX5474932_ERR5766176_1.fastq.gz, /nf-core/test-datasets/taxprofiler/data/fastq/ERX5474932_ERR5766176_2.fastq.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'2612', 'run_accession':'ERR5766176_B', 'instrument_platform':'ILLUMINA', 'single_end':false, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERX5474932_ERR5766176_B_1.fastq.gz, /nf-core/test-datasets/taxprofiler/data/fastq/ERX5474932_ERR5766176_B_2.fastq.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'2612', 'run_accession':'ERR5766180', 'instrument_platform':'ILLUMINA', 'single_end':true, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERX5474936_ERR5766180_1.fastq.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'2613', 'run_accession':'ERR5766181', 'instrument_platform':'ILLUMINA', 'single_end':false, 'is_fasta':false, 'type':'short', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERX5474937_ERR5766181_1.fastq.gz, /nf-core/test-datasets/taxprofiler/data/fastq/ERX5474937_ERR5766181_2.fastq.gz]]
[DUMP: diamond_test-noprocessing_reads] [['id':'ERR3201952', 'run_accession':'ERR3201952', 'instrument_platform':'OXFORD_NANOPORE', 'single_end':true, 'is_fasta':false, 'type':'long', 'tool':'diamond', 'db_name':'db1', 'db_params':''], [/nf-core/test-datasets/taxprofiler/data/fastq/ERR3201952.fastq.gz]]
```

Comment (jfy133) [reply subworkflows/local/profiling.nf:353]: @LilyAnderssonLee what does the resolvd `.command.sh` look like?

**Comment by LilyAnderssonLee** [comment]
Here is the error message and no `command.sh` file:
```
Caused by:
  No signature of method: nextflow.util.BlankSeparatedList.getExtension() is applicable for argument types: () values: [] -- Check script './workflows/../subworkflows/local/../../modules/nf-core/diamond/blastx/main.nf' at line: 33

Source block:
  def args = task.ext.args ?: ''
  def prefix = task.ext.prefix ?: "${meta.id}"
  def is_compressed = fasta.getExtension() == "gz" ? true : false
  def fasta_name = is_compressed ? fasta.getBaseName() : fasta
  def columns = blast_columns ? "${blast_columns}" : ''
  switch ( out_ext ) {
          case "blast": outfmt = 0; break
          case "xml": outfmt = 5; break
          case "txt": outfmt = 6; break
          case "daa": outfmt = 100; break
          case "sam": outfmt = 101; break
          case "tsv": outfmt = 102; break
          case "paf": outfmt = 103; break
          default:
              outfmt = '6';
              out_ext = 'txt';
              log.warn("Unknown output file format provided (${out_ext}): selecting DIAMOND default of tabular BLAST output (txt)");
              break
      }
  """
      if [ "${is_compressed}" == "true" ]; then
          gzip -c -d ${fasta} > ${fasta_name}
      fi
  
      DB=`find -L ./ -name "*.dmnd" | sed 's/\\.dmnd\$//'`
  
      diamond \\
          blastx \\
          --threads ${task.cpus} \\
          --db \$DB \\
          --query ${fasta_name} \\
          --outfmt ${outfmt} ${columns} \\
          ${args} \\
          --out ${prefix}.${out_ext} \\
          --log
  
      mv diamond.log ${prefix}.log
  
      cat <<-END_VERSIONS > versions.yml
      "${task.process}":
          diamond: \$(diamond --version 2>&1 | tail -n 1 | sed 's/^diamond version //')
      END_VERSIONS
      """

```

File: subworkflows/local/profiling.nf (hunk 1 above)
Comment (LilyAnderssonLee) [reply subworkflows/local/profiling.nf:353]: Here is the error message and no `command.sh` file:
```
Caused by:
  No signature of method: nextflow.util.BlankSeparatedList.getExtension() is applicable for argument types: () values: [] -- Check script './workflows/../subworkflows/local/../../modules/nf-core/diamond/blastx/main.nf' at line: 33

Source block:
  def args = task.ext.args ?: ''
  def prefix = task.ext.prefix ?: "${meta.id}"
  def is_compressed = fasta.getExtension() == "gz" ? true : false
  def fasta_name = is_compressed ? fasta.getBaseName() : fasta
  def columns = blast_columns ? "${blast_columns}" : ''
  switch ( out_ext ) {
          case "blast": outfmt = 0; break
          case "xml": outfmt = 5; break
          case "txt": outfmt = 6; break
          case "daa": outfmt = 100; break
          case "sam": outfmt = 101; break
          case "tsv": outfmt = 102; break
          case "paf": outfmt = 103; break
          default:
              outfmt = '6';
              out_ext = 'txt';
              log.warn("Unknown output file format provided (${out_ext}): selecting DIAMOND default of tabular BLAST output (txt)");
              break
      }
  """
      if [ "${is_compressed}" == "true" ]; then
          gzip -c -d ${fasta} > ${fasta_name}
      fi
  
      DB=`find -L ./ -name "*.dmnd" | sed 's/\\.dmnd\$//'`
  
      diamond \\
          blastx \\
          --threads ${task.cpus} \\
          --db \$DB \\
          --query ${fasta_name} \\
          --outfmt ${outfmt} ${columns} \\
          ${args} \\
          --out ${prefix}.${out_ext} \\
          --log
  
      mv diamond.log ${prefix}.log
  
      cat <<-END_VERSIONS > versions.yml
      "${task.process}":
          diamond: \$(diamond --version 2>&1 | tail -n 1 | sed 's/^diamond version //')
      END_VERSIONS
      """

```

**Comment by jfy133** [comment]
To summarise the issue:

- DIAMOND doesn't actually accept paired end input (Silent fail in taxprofiler <= 1.1.8, it would just take the first file)
- A new check in the module expects a single file not a list, but we sometimes send pairs, and so `get.Extesion()` doesn't work if there is >= 1 file
- With nopreprocessing there is no read-merging, so remains as pairs rather than a single file

@LilyAnderssonLee and my current thinking:

- Have a warning that if paried-end data is input + diamond turned on, give a warning it will be filtered out
- Filter out paire-dend data before it goes into diamond

**Comment by LilyAnderssonLee** [comment]
@jfy133 The `multiqc` module has been updated to the latest version, but `ganon` can't be included in the MultiQC report because the log file generated by the latest `ganon` version 2.0.0 differs from the one generated by version 1.5.1 used in the MultiQC ganon module.

**Comment by LilyAnderssonLee** [comment]
`nanoq` has been added MultiQC report. 
[multiqc_report.html.zip](https://github.com/user-attachments/files/16814859/multiqc_report.html.zip)

**Comment by LilyAnderssonLee** [comment]
All modules have been updated except for `motus/profile`. I have opened a [PR](https://github.com/nf-core/modules/pull/6212) to update this module. But there are some issues running `nf-test `at the global setup `MOTUS_DOWNLOADDB` due to the large database. I believe we need to prepare a smaller test database for `motus`, which will take some time. Since Sofia is updating `motus/profile` for long reads, I’m considering if we could skip the update for `motus/profile` for now and include it in the third release. This way, we could proceed with the release this week. What do you think @nf-core/taxprofiler ?

**Comment by jfy133** [comment]
I had one last idea for motus/profile, so testing that otherwise yes we can skip that one for now

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,11 +12,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+| bbmap     | 39.01            | 39.06       |
+| bowtie2   | 2.4.4            | 2.5.2       |
+| bracken   | 2.7              | 2.9         |
+| diamond   | 2.0.15           | 2.1.8       |
+| ganon     | 1.5.1            | 2.0.0       |
+| kraken2   | 2.1.2            | 2.1.3       |
+| minimap2  | 2.24             | 2.28        |
+| samtools  | 1.17             | 1.20        |
+| megan     | 6.24.20          | 6.25.9      |
+| metaphlan | 4.0.6            | 4.1.1       |
```
Comment (sofstam) [inline CHANGELOG.md:34]: Add motus,blastx, gunzip,krona,malt,multiqc,untar  new version as well.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,11 +12,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+| --------- | ---------------- | ----------- |
+| bbmap     | 39.01            | 39.06       |
+| bowtie2   | 2.4.4            | 2.5.2       |
+| bracken   | 2.7              | 2.9         |
+| diamond   | 2.0.15           | 2.1.8       |
+| ganon     | 1.5.1            | 2.0.0       |
+| kraken2   | 2.1.2            | 2.1.3       |
+| minimap2  | 2.24             | 2.28        |
+| samtools  | 1.17             | 1.20        |
+| megan     | 6.24.20          | 6.25.9      |
```
Comment (sofstam) [inline CHANGELOG.md:33]: Was it 6.21.7 before?

File: CHANGELOG.md (hunk 2 above)
Comment (sofstam) [reply CHANGELOG.md:34]: Shall nanoq be included?

File: subworkflows/local/longread_hostremoval.nf [hunk 4]
```diff
@@ -20,13 +20,13 @@ workflow LONGREAD_HOSTREMOVAL {
     ch_multiqc_files  = Channel.empty()
 
     if ( !params.longread_hostremoval_index ) {
-        ch_minimap2_index = MINIMAP2_INDEX ( [ [], reference ] ).index.map { it[1] }
+        ch_minimap2_index = MINIMAP2_INDEX ( [ [], reference ] ).index
```
Comment (sofstam) [inline subworkflows/local/longread_hostremoval.nf:23]: Why was this removed?

**Review by sofstam — COMMENTED** [review]
Shall it be documented in the usage that Nanoq is going to be the default option if a user chooses filtering?

File: modules/nf-core/megan/rma2info/main.nf [hunk 5]
```diff
@@ -2,10 +2,10 @@ process MEGAN_RMA2INFO {
     tag "$meta.id"
     label 'process_single'
 
-    conda 'modules/nf-core/megan/rma2info/environment.yml'
+    conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/megan:6.24.20--h9ee0642_0':
```
Comment (LilyAnderssonLee) [inline modules/nf-core/megan/rma2info/main.nf:7]: @sofstam Regarding your comments about malt version change, the container pulled version 6.24.20 here.

File: subworkflows/local/longread_hostremoval.nf (hunk 4 above)
Comment (LilyAnderssonLee) [reply subworkflows/local/longread_hostremoval.nf:23]: Due to the the input reference channel change in the process `MINIMAP2_ALIGN`. The new module needs meta for index: `tuple val(meta2), path(reference)`

File: CHANGELOG.md (hunk 2 above)
Comment (LilyAnderssonLee) [reply CHANGELOG.md]: Updated!

File: docs/usage.md [hunk 6]
```diff
@@ -303,7 +303,7 @@ Complexity filtering is primarily a run-time optimisation step. It is not necess
 
 There are currently three options for short-read complexity filtering: [`bbduk`](https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/), [`prinseq++`](https://github.com/Adrian-Cantu/PRINSEQ-plus-plus), and [`fastp`](https://github.com/OpenGene/fastp#low-complexity-filter).
 
-There are two options for long-read quality filtering: [`Filtlong`](https://github.com/rrwick/Filtlong) and [`nanoq`](https://github.com/esteinig/nanoq).
+There are two options for long-read quality filtering: [`Filtlong`](https://github.com/rrwick/Filtlong) and [`nanoq`](https://github.com/esteinig/nanoq), with `nanoq` being the default option.
```
Comment (sofstam) [inline docs/usage.md:306]:
```suggestion
There are two options for long-read quality filtering: [`Filtlong`](https://github.com/rrwick/Filtlong) and [`Nanoq`](https://github.com/esteinig/nanoq), with `Nanoq` being the default option.
```

Comment (LilyAnderssonLee) [reply docs/usage.md:306]: I prefer `nanoq`. That's the default name from the https://github.com/esteinig/nanoq

File: CHANGELOG.md [hunk 7]
```diff
@@ -12,11 +12,33 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- [#518](https://github.com/nf-core/taxprofiler/pull/518) Fixed a bug where Oxford Nanopore FASTA input files would not be processed (❤️ to @ikarls for reporting, fixed by @jfy133)
+
 ### `Dependencies`
 
+| Tool                     | Previous version | New version |
+| ------------------------ | ---------------- | ----------- |
+| bbmap                    | 39.01            | 39.06       |
+| bowtie2                  | 2.4.4            | 2.5.2       |
+| bracken                  | 2.7              | 2.9         |
+| cat/fastq:coreutils=8.30 |                  | 8.30        |
```
Comment (sofstam) [inline CHANGELOG.md:28]: What about 

```suggestion
| cat/fastq |                  | 8.30        |
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -12,11 +12,33 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+| kraken2                  | 2.1.2            | 2.1.3       |
+| krona                    | 2.8              | 2.8.1       |
+| megan                    | 6.24.20          | 6.25.9      |
+| metaphlan                | 4.0.6            | 4.1.1       |
+| minimap2                 | 2.24             | 2.28        |
+| motus/profile            | 3.0.3            | 3.1.0       |
+| multiqc                  | 1.21             | 1.24.1      |
+| nanoq                    |                  | 0.10.0      |
+| samtools                 | 1.17             | 1.20        |
+| untar:sed                | 4.7              | 4.8         |
```
Comment (sofstam) [inline CHANGELOG.md:40]: Same here

File: assets/schema_input.json [hunk 9]
```diff
@@ -50,6 +50,7 @@
                 "type": "string",
                 "format": "file-path",
                 "pattern": "^\\S+\\.(f(ast)?q|fa(sta)?)\\.gz$",
+                "unique": true,
```
Comment (sofstam) [inline assets/schema_input.json:53]: ⭐

File: CHANGELOG.md (hunk 7 above)
Comment (LilyAnderssonLee) [reply CHANGELOG.md:28]: I am not sure. Since coreutils=8.30 was used in the module. If you think it's proper, I could update.

File: CHANGELOG.md (hunk 8 above)
Comment (LilyAnderssonLee) [reply CHANGELOG.md:40]: If I recall correctly, sed was also used somewhere else and version is still 4.7. Do you think I should change `untar: sed` to `sed`

File: CHANGELOG.md (hunk 7 above)
Comment (sofstam) [reply CHANGELOG.md:28]: Maybe just `cat/fastq` is enough?

File: CHANGELOG.md (hunk 8 above)
Comment (sofstam) [reply CHANGELOG.md:40]: I think we should use the version of untar.

Comment (LilyAnderssonLee) [reply CHANGELOG.md:40]: Ok.  I will update it.

**Comment by jfy133** [comment]
@LilyAnderssonLee I think we want unique file names for ALL types of files (fastq/fasta) etc!

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Comment by LilyAnderssonLee** [comment]
> @LilyAnderssonLee I think we want unique file names for ALL types of files (fastq/fasta) etc!

Initially, I thought it wasn't necessary since, for FASTQ files, the reads channel from preprocessing or host removal is defined as `${meta.id}_${meta.run_accession}` and `run_accession` is unique. But if someone wants to run profiling directly that could be a problem. I suppose I could add `unique:true` to the FASTQ files.

**Comment by jfy133** [comment]
> Initially, I thought it wasn't necessary since, for FASTQ files, the reads channel from preprocessing or host removal is defined as `${meta.id}_${meta.run_accession}` which is unique. But if someone wants to run profiling directly that could be a problem. I suppose I could add `unique:true` to the FASTQ files.

You're right it's not so bad after the first couple of steps as indeed we come up with our own names.

But for the initial steps it might be safer, and also (now I think about it) , it would help prevent any copy pasting errors when creating the samplesheet too so I think it would be worth it :).

If someone has the exact same file name for different samples or libraries on their system, this is dangerous for them too so we can nudge them to also be more careful with that 😬

**Review by sofstam — APPROVED** [review]
